### ISSUE TITLE
[BUG] npm v8.11.0 not seeing locally installed package when running versioned command

### ISSUE THREAD

**Issue by shayneczyzewski** [npm/cli#5018]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### This issue exists in the latest npm version

- [X] I am using the latest npm

### Current Behavior

With a clean npm/npx environment and the corresponding package installed locally in node_modules, when running a versioned `npm exec`/`npx` command, like: `npm exec prisma@3.9.1`, v8.5.5 finds it in the project and does _not_ ask to install, but v8.11.0 asks you to install it, like so:

<img width="1002" alt="Screen Shot 2022-06-14 at 10 42 52 AM" src="https://user-images.githubusercontent.com/523636/173606002-545c2df6-6f48-4235-819d-0369d1807847.png">

### Expected Behavior

I would expect it to behave like v8.5.5, find the locally installed version, and not request you install it, like so:

<img width="892" alt="Screen Shot 2022-06-14 at 10 43 59 AM" src="https://user-images.githubusercontent.com/523636/173606321-a16dba92-c49b-42ff-aa40-6826dd927b5f.png">

### Steps To Reproduce

1. In several different environments tested so far...
2. Clone this sample repo: https://github.com/shayneczyzewski/npm-prisma-repro
3. Clear your npm (`npm cache clean --force`) and npm (`rm -rf ~/.npm/_npx`) caches
4. Run `npm install` in the project
5. `nvm use 16.15.0` and observe output of `npm exec prisma@3.9.1` is the help command [Expected]
6. `nvm use 16.15.1` and observe output of `npm exec prisma@3.9.1` is a request to install

### Environment

- npm: v8.11.0
- Node.js: v16.15.1
- OS Name: macOS Monterey (Version 12.4)
- System Model Name: MacBook Air M1 2020
- npm config:
```ini
; node bin location = /Users/shayne/.nvm/versions/node/v16.15.1/bin/node
; node version = v16.15.1
; npm local prefix = /Users/shayne/dev/issueRepro/node-16.15.1-failure
; npm version = 8.11.0
; cwd = /Users/shayne/dev/issueRepro/node-16.15.1-failure
; HOME = /Users/shayne
; Run `npm config ls -l` to show all defaults.
```

**Comment by shayneczyzewski** [comment]
Hi there, is anyone available to help triage this? Happy to help however I can. Thanks so much.

**Comment by sodic** [comment]
Hello, I can reproduce the bug on the following Linux machine:
- OS Name: Manjaro Linux x86_64 (5.15.48-1-MANJARO)
- System Model Name: Dell XPS 15 9510
- Output of `npm config ls`:
```userconfig = "/home/filip/.config/npm/npmrc" 

; node bin location = /home/filip/.local/share/nvm/versions/node/v16.15.1/bin/node
; node version = v16.15.1
; npm local prefix = /tmp/npm-prisma-repro
; npm version = 8.11.0
; cwd = /tmp/npm-prisma-repro
; HOME = /home/filip
; Run `npm config ls -l` to show all defaults.
```

I can also confirm the bug is not package-specific. I successfully reproduced it with several packages other than Prisma.

I'd love to see this addressed. Also happy to help in any way!

**Comment by shayneczyzewski** [comment]
Hi there @lukekarrys sorry to disturb, but I haven't gotten any feedback since this issue was submitted and I was wondering if there may be anything else I can do to help the cli team triage this potentially large issue. Thanks so much for your guidance!

**Comment by wraithgar** [comment]
This should be fixed as of `npm@8.16.0`

```sh
~/D/n/s/npx $ npm ls json
npx@1.0.0 /Users/wraithgar/Development/npm/scratch/npx
└── json@9.0.6

~/D/n/s/npx $ npx json@9 --version
json 9.0.6
written by Trent Mick
https://github.com/trentm/json

```

**Comment by shayneczyzewski** [comment]
Thanks, @wraithgar, appreciate it!
